### DISCUSSION TITLE
Interpretation of Sankey diagrams

### DISCUSSION THREAD

**Discussion by dbantje** [LCA-ActivityBrowser/activity-browser#1392] [Q&A] [answered]
Hi all!

I'm having trouble understanding the Sankey diagram, in particular the impact fractions shown for edges in the diagram.

![Capture_sankey1](https://github.com/user-attachments/assets/9db4e15a-e915-46c8-b2db-4960b9d963f0)

In the subgraph above, how can I make sense of the fact that the top edge has a higher associated impact than the downstream one? I would have expected that downstream edges must always have a larger associated impact.

Thanks!

**Comment by marc-vdm** [comment, marked as answer]
Hi, The Sankey works by calculating an LCA for each visible activity (node) in the graph, scaled to how much of that activity is used by the previous one. This may sound a bit abstract, but I hope the simple example on the [Sankey wiki page](https://github.com/LCA-ActivityBrowser/activity-browser/wiki/LCA-Results#sankey) helps clarify.

As systems can become _very_ large (e.g. ecoinvent has >20000 activities),  the graph traversal is cut-off at some amount of steps (250 by default), or when the impact becomes lower than some percentage of the total score (5% by default), whichever comes first.

One explanation for what you are seeing could be that the `market for scrap copper` also has a negative impact for your impact category that is not shown because one of the cutoff. You could calculate the results for the `market for scrap copper` activity and explore why your results are what they are.

Hope this helps!

**Reply by marc-vdm** [reply]
Hi @dbantje, if the above resolved your question, please `Mark as answer` so other users with the same question can more easily find the answer :)

**Reply by nicolnt** [reply]
@marc-vdm is the "amount of steps" referring to the number of recursive analysis or the absolute number of calculations?
By that, I mean if I set cutoff 2, is it supposed to show the dependencies of the dependencies (2 recursive call) or just two processes? It's not clear in the [wiki](https://github.com/LCA-ActivityBrowser/activity-browser/wiki/LCA-Results#sankey-configuration) and from my own testings.

**Reply by marc-vdm** [reply]
Hi @nicolnt,

According to the [brightway documentation](https://docs.brightway.dev/en/legacy/content/api/bw2calc/graph_traversal/index.html#module-bw2calc.graph_traversal), the 'amount of steps' is based on the amount of activities. However, I do believe the graph traversal may sometimes calculate a few more processes, though it's not entirely clear to me why.

**Reply by nicolnt** [reply]
OK thanks, I thought it was something handled by AB, but I will check this more in depth in Brightway. 
If possible, can you point me where in AB code those functions are used? Otherwise, I will attempt to figure this out by myself.
Best,

**Reply by marc-vdm** [reply]
You can probably best start your search here:
https://github.com/LCA-ActivityBrowser/activity-browser/blob/d415e95615bb63601a9b954755eab6766e8211d3/activity_browser/ui/web/sankey_navigator.py#L230-L298

**Comment by nicolnt** [comment]
This higher value of _scrap copper_ from _treatment of scrap copper municipal incineration RoW_ could be explained if _market for scrap copper RoW_ is consumed by other activities than _distribution network construction, electricity, low voltage RoW_ in the system for which the flows are smaller than the cutoff, which is highly possible ([ecoquery ecoinvent 3.9 cutoff](https://ecoquery.ecoinvent.org/3.9/cutoff/dataset/14783/consuming_activities)).

Therefore, the missing ~4.1% could be the sum of many exchanges taking place across the system, not being shown due to a combination of the cutoff value and the depth.
